max-wave.lua

```diff
@@ -0,0 +1,63 @@
+--Dynamically limit the next immigration wave
+--By Loci, modified by Fleeting Frames and Tachytaenius
+--[====[
```

Comment:
some things have changed since this PR started. Docs now go in the `docs/` folder. You can copy an existing file in there to get the current formats, or read the new documentation guidance here: https://docs.dfhack.org/en/latest/docs/Documentation.html

Reply:
Let me know if the docs I pushed are OK.